lib/editor/helpers.js

```diff
@@ -20,21 +20,39 @@ export function getBufferPositionFromMouseEvent(event: MouseEvent, editor: TextE
   return null
 }
 
-export function mouseEventNearPosition(event: MouseEvent, editorElement: Object, screenPosition: Point, elementWidth: number, elementHeight: number): boolean {
+export function mouseEventNearPosition({ event, editor, editorElement, tooltipElement, screenPosition }: Object): boolean {
   const pixelPosition = editorElement.component.pixelPositionForMouseEvent(event)
   const expectedPixelPosition = editorElement.pixelPositionForScreenPosition(screenPosition)
   const differenceTop = pixelPosition.top - expectedPixelPosition.top
   const differenceLeft = pixelPosition.left - expectedPixelPosition.left
-  if (differenceTop === 0 && differenceLeft === 0) {
-    return true
-  }
-  if ((differenceTop > 0 && differenceTop > (elementHeight + 20)) || (differenceTop < 0 && differenceTop < -5)) {
-    return false
+
+  const editorLineHeight = editor.lineHeightInPixels
+  const elementHeight = tooltipElement.offsetHeight + editorLineHeight
+  const elementWidth = tooltipElement.offsetWidth
+
+  if (differenceTop > 0) {
+    // Cursor is below the line
+    if (differenceTop > (elementHeight + (1.5 * editorLineHeight))) {
+      return false
+    }
+  } else if (differenceTop < 0) {
+    // Cursor is above the line
+    if (differenceTop < (-1.5 * editorLineHeight)) {
+      return false
+    }
   }
-  if (differenceLeft > 15 && differenceTop < 17) {
-    return false
+  if (differenceLeft > 0) {
+    // Right of the start of highlight
+    if (differenceLeft > (elementWidth + 30)) {
```

Comment:
Can a character width be used here maybe? Might be nice to move away from arbitrary numbers entirely.